presentation/frontend/src/presentation-frontend/selection/SelectionManager.ts

```diff
@@ -635,25 +639,31 @@ class IModelSelectionStorage {
     });
   }
 
-  public computeSelection(level: number, selectables: Selectables) {
+  public computeSelection(level: number, currSelectables: Selectables, changedSelectables: Selectables) {
     this.clearSelections(level);
 
     const prevComputationsDisposers = [...(this._currentSelection.get(level)?.ongoingComputationDisposers ?? [])];
-    const thisDisposer = new Subject<void>();
-    this.addDisposer(level, thisDisposer);
+    const currDisposer = new Subject<void>();
+    this.addDisposer(level, currDisposer);
 
     return defer(async () => {
-      const keys = await selectablesToKeys(selectables);
+      const currentSelectionResult = await selectablesToKeys(currSelectables);
+      const changedSelectionResult = await selectablesToKeys(changedSelectables, currentSelectionResult.selectableKeys);
```

Comment:
any reason not to use `Promise.all` here?

Reply:
Second call of `selectablesToKeys` uses result of first call to avoid loading instance keys for same custom selectable twice.

Reply:
could they both take an array of handled keys and add items there as the keys are handled?